cmd/waku/rpc/relay.go

```diff
@@ -78,18 +85,34 @@ func (r *RelayService) Start() {
 	r.runner.Start()
 }
 
+// Stop stops the RelayService
 func (r *RelayService) Stop() {
 	r.runner.Stop()
 }
 
+// PostV1Message is invoked when the json rpc request uses the post_waku_v2_relay_v1_message method
 func (r *RelayService) PostV1Message(req *http.Request, args *RelayMessageArgs, reply *SuccessReply) error {
 	var err error
 
-	if args.Topic == "" {
-		_, err = r.node.Relay().Publish(req.Context(), args.Message.toProto())
-	} else {
-		_, err = r.node.Relay().PublishToTopic(req.Context(), args.Message.toProto(), args.Topic)
+	topic := relay.DefaultWakuTopic
+	if args.Topic != "" {
+		topic = args.Topic
 	}
+
+	if !r.node.Relay().IsSubscribed(topic) {
+		return errors.New("not subscribed to pubsubTopic")
+	}
+
+	msg := args.Message.toProto()
+
+	if node.SupportsRLN {
+		err = r.node.RLNRelay().AppendRLNProof(msg, r.node.Timesource().Now())
+		if err != nil {
+			return err
```

Comment:
It would be better to append this error to something like "unable to publish to topic due to RLN error: <errstring>"